File: Grace/Grace.c
#include <stdio.h>
#include <fcntl.h>
#define OPEN(x) fopen(x, "w")
#define STRING() char*
#define FT() int main(){STRING() str="#include <stdio.h>%c#include <fcntl.h>%c#define OPEN(x) fopen(x, %cw%c)%c#define STRING() char*%c#define FT() int main(){STRING() str=%c%s%c;FILE *kid=OPEN(%cGrace_kid.c%c);if(kid)fprintf(kid, str, 10, 10, 34, 34, 10, 10, 34, str, 34, 34, 34, 10, 10, 10, 9, 10, 10);}%cFT()%c/*%c%cComment%c*/%c";FILE *kid=OPEN("Grace_kid.c");if(kid)fprintf(kid, str, 10, 10, 34, 34, 10, 10, 34, str, 34, 34, 34, 10, 10, 10, 9, 10, 10);}
FT()
/*
	Comment
*/

File: bonus/Grace.py
def CLOSE(file): file.close()
def WRITE(file, str): file.write(str)
def MACRO(): str="def CLOSE(file): file.close()%cdef WRITE(file, str): file.write(str)%cdef MACRO(): str=%c%s%c;file=open(%cGrace_kid.py%c, %cw%c);WRITE(file, str%% (10, 10, 34, str, 34, 34, 34, 34, 34, 10, 10, 10, 10));CLOSE(file)%c%cMACRO()%c#comment%c";file=open("Grace_kid.py", "w");WRITE(file, str% (10, 10, 34, str, 34, 34, 34, 34, 34, 10, 10, 10, 10));CLOSE(file)

MACRO()
#comment
